### PR TITLE
MGMT-9555 - Add more complex triggers conditions comparison methods

### PR DESCRIPTION
**Pull request by eliorerz**
Add the ability to not just activate trigger on "equal" condition, but also define the condition as lambda 
/cc @osherdp

### CODE REVIEW

File: src/triggers/condition.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any
+
+
+class TriggerConditionType(Enum):
```
Comment (osherdp) [inline src/triggers/condition.py:6]:
```suggestion
class TriggerConditionOperator(Enum):
```

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/triggers/default_triggers.py [hunk 2]
```diff
@@ -3,26 +3,21 @@
...
-        "production": Trigger(
-            condition=("remote_service_url", consts.RemoteEnvironment.PRODUCTION), worker_disk=consts.DISK_SIZE_120GB
-        ),
-        "staging": Trigger(
-            condition=("remote_service_url", consts.RemoteEnvironment.STAGING), worker_disk=consts.DISK_SIZE_120GB
-        ),
-        "integration": Trigger(
-            condition=("remote_service_url", consts.RemoteEnvironment.INTEGRATION), worker_disk=consts.DISK_SIZE_120GB
+        "remote_deployment": Trigger(
+            Condition("remote_service_url", None, TriggerConditionType.NOT_EQUAL), worker_disk=consts.DISK_SIZE_120GB
```
[the remark this answers is not in the document]
Comment (eliorerz) [reply src/triggers/default_triggers.py]: I think you might be right, I'll remove that.

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Pull request merged**